False positive in F541 (f-string without placeholders) after v0.0.204

not sure if it's #1494 or a different PR... but a F541 "f-string without placeholders" false positive is now appearing in v0.0.204 when using format specifiers in f-strings:

```python
v = 23.234234
f"{v:0.2f}"  # error
```

Another reproducer:

```
(sphinx) PS S:\Development\sphinx> type bug.py                
f' style="vertical-align: {-0:d}px"'
(sphinx) PS S:\Development\sphinx> pip install "ruff==0.0.203" --quiet
(sphinx) PS S:\Development\sphinx> ruff -V
ruff 0.0.203
(sphinx) PS S:\Development\sphinx> ruff --select F bug.py             
(sphinx) PS S:\Development\sphinx> pip install "ruff==0.0.204" --quiet
(sphinx) PS S:\Development\sphinx> ruff -V                            
ruff 0.0.204
(sphinx) PS S:\Development\sphinx> ruff --select F bug.py             
bug.py:1:1: F541 f-string without any placeholders
Found 1 error(s).
```

A

--- a/resources/test/fixtures/pyflakes/F541.py
+++ b/resources/test/fixtures/pyflakes/F541.py
@@ -1,6 +1,8 @@
+# OK
 a = "abc"
 b = f"ghi{'jkl'}"
 
+# Errors
 c = f"def"
 d = f"def" + "ghi"
 e = (
@@ -8,5 +10,17 @@ e = (
     "ghi"
 )
 
+# OK
 g = f"ghi{123:{45}}"
+
+# Error
 h = "x" "y" f"z"
+
+v = 23.234234
+
+# OK
+f"{v:0.2f}"
+
+# OK (false negatives)
+f"{v:{f'0.2f'}}"
+f"{f''}"

--- a/src/checkers/ast.rs
+++ b/src/checkers/ast.rs
@@ -91,6 +91,7 @@ pub struct Checker<'a> {
     in_type_definition: bool,
     in_deferred_string_type_definition: bool,
     in_deferred_type_definition: bool,
+    in_f_string: bool,
     in_literal: bool,
     in_subscript: bool,
     seen_import_boundary: bool,
@@ -147,6 +148,7 @@ impl<'a> Checker<'a> {
             in_type_definition: false,
             in_deferred_string_type_definition: false,
             in_deferred_type_definition: false,
+            in_f_string: false,
             in_literal: false,
             in_subscript: false,
             seen_import_boundary: false,
@@ -1466,6 +1468,7 @@ where
 
         self.push_expr(expr);
 
+        let prev_in_f_string = self.in_f_string;
         let prev_in_literal = self.in_literal;
         let prev_in_type_definition = self.in_type_definition;
 
@@ -2151,7 +2154,9 @@ where
                 }
             }
             ExprKind::JoinedStr { values } => {
-                if self.settings.enabled.contains(&CheckCode::F541) {
+                // Conversion flags are parsed as f-strings without placeholders, so skip
+                // nested f-strings, which would lead to false positives.
+                if !self.in_f_string && self.settings.enabled.contains(&CheckCode::F541) {
                     if !values
                         .iter()
                         .any(|value| matches!(value.node, ExprKind::FormattedValue { .. }))
@@ -2654,6 +2659,11 @@ where
                 }
                 self.in_subscript = prev_in_subscript;
             }
+            ExprKind::JoinedStr { .. } => {
+                self.in_f_string = true;
+                visitor::walk_expr(self, expr);
+                self.in_f_string = prev_in_f_string;
+            }
             _ => visitor::walk_expr(self, expr),
         }
 
@@ -2671,6 +2681,7 @@ where
 
         self.in_type_definition = prev_in_type_definition;
         self.in_literal = prev_in_literal;
+        self.in_f_string = prev_in_f_string;
 
         self.pop_expr();
     }

